packages/core/getBusyTimes.ts

```diff
@@ -34,43 +34,97 @@ export async function getBusyTimes(params: {
       status: BookingStatus.ACCEPTED,
     })}`
   );
+
+  /**
+   * A user is considered busy within a given time period if there
+   * is a booking they own OR host.
+   *
+   * Therefore this query does the following:
+   * - Performs a query for all EventType id's where this user is a host
+   * - Performs a query for all bookings where:
+   *   - The given booking is owned by this user, or..
+   *   - The given booking's EventType is hosted by this user
+   *
+   * See further discussion within this GH issue:
+   * https://github.com/calcom/cal.com/issues/6374
+   *
+   * NOTE: Changes here will likely require changes to some mocking
+   *  logic within getSchedule.test.ts:addBookings
+   */
   performance.mark("prismaBookingGetStart");
-  const busyTimes: EventBusyDetails[] = await prisma.booking
-    .findMany({
-      where: {
-        userId,
-        startTime: { gte: new Date(startTime) },
-        endTime: { lte: new Date(endTime) },
-        status: {
-          in: [BookingStatus.ACCEPTED],
+  const busyTimes: EventBusyDetails[] =
```

Comment:
Excellent going! Needs one amend though as currently if you are not the host of any event type (non-team user) `prisma.host.findMany` will return an empty array, this will cause all following then's to be skipped, in which case the bookings owned by the user are no longer included.